hibernate-reactive/src/main/kotlin/com/linecorp/kotlinjdsl/query/HibernateStageReactiveQuery.kt

```diff
@@ -0,0 +1,56 @@
+package com.linecorp.kotlinjdsl.query
+
+import org.hibernate.reactive.stage.Stage
+import java.util.concurrent.CompletionStage
+import javax.persistence.Parameter
+import kotlin.reflect.KClass
+
+class HibernateStageReactiveQuery<R>(private val query: Stage.Query<R>) : ReactiveQuery<R> {
+    override val singleResult: CompletionStage<R>
+        get() = query.singleResult
+    override val resultList: CompletionStage<List<R>>
+        get() = query.resultList
+    override val singleResultOrNull: CompletionStage<R?>
+        get() = query.singleResultOrNull
+    override val executeUpdate: CompletionStage<Int>
+        get() = query.executeUpdate()
+
+    override fun setParameter(position: Int, value: Any?): ReactiveQuery<R> {
+        query.setParameter(position, value)
+        return this
+    }
+
+    override fun setParameter(name: String, value: Any?): ReactiveQuery<R> {
+        query.setParameter(name, value)
+        return this
+    }
+
+    override fun <T> setParameter(parameter: Parameter<T>, value: T?): ReactiveQuery<R> {
+        query.setParameter(parameter, value)
+        return this
+    }
+
+    override fun setMaxResults(maxResults: Int): ReactiveQuery<R> {
+        query.maxResults = maxResults
+        return this
+    }
+
+    override fun setFirstResult(firstResult: Int): ReactiveQuery<R> {
+        query.firstResult = firstResult
+        return this
+    }
+
+    override fun setQueryHint(hintName: String, value: Any) {
+        // hint not supported in hibernate-reactive
```

Comment:
-- english
Since Hibernate does not support it, it would be better to throw an exception. Or it'd be good to log it as warn.
If there is only comment, the user will recognize it as a bug because the QueryHint operation does not work.

-- korean
지원하지 않기 때문에 에러를 발생시키는 것은 어떤가요? 아니면 warn 로그로 남기는 것도 좋을 것 같습니다.
주석만 적을 경우 사용자가 QueryHint 동작이 되지 않아 버그로 인식할 것 같습니다.

Reply:
applied

-- korean
반영했습니다.